packages/client/test/unit/GapFiller.test.ts

```diff
@@ -0,0 +1,241 @@
+import { MessageID, MessageRef, StreamMessage, StreamPartIDUtils } from '@streamr/protocol'
+import { Defer, EthereumAddress, toEthereumAddress, wait, waitForCondition } from '@streamr/utils'
+import { OrderedMessageChain } from '../../src/subscribe/ordering/OrderedMessageChain'
+import { GapFiller } from '../../src/subscribe/ordering/GapFiller'
+import { Gap } from '../../src/subscribe/ordering/OrderedMessageChain'
+import { fromArray } from '../../src/utils/GeneratorUtils'
+
+const CONTEXT = {
+    streamPartId: StreamPartIDUtils.parse('stream#0'),
+    publisherId: toEthereumAddress('0x0000000000000000000000000000000000000001'),
+    msgChainId: 'msgChainId'
+}
+const STORAGE_NODE_ADDRESS = toEthereumAddress('0x0000000000000000000000000000000000000002')
+
+const createMessage = (timestamp: number, hasPrevRef = true) => {
+    return new StreamMessage({
+        messageId: new MessageID(
+            StreamPartIDUtils.getStreamID(CONTEXT.streamPartId),
+            StreamPartIDUtils.getStreamPartition(CONTEXT.streamPartId),
+            timestamp,
+            0,
+            CONTEXT.publisherId, 
+            CONTEXT.msgChainId
+        ),
+        prevMsgRef: hasPrevRef ? new MessageRef(timestamp - 1, 0) : null,
+        content: '{}',
+        signature: 'signature'
+    })
+}
+
+const MAX_REQUESTS_PER_GAP = 3
+const INITIAL_WAIT_TIME = 50
+
+describe('GapFiller', () => {
+
+    let chain: OrderedMessageChain
+    const onOrderedMessageAdded = jest.fn()
```

Comment:
```suggestion
    let onOrderedMessageAdded
```
This should be set in `beforeEach` so it is reset for each test?